src/lib/bootstrap_controller/bootstrap_controller.mli

```diff
@@ -2,10 +2,20 @@ open Async_kernel
 open Pipe_lib
 open Network_peer
 
+module type CONTEXT = sig
+  val logger : Logger.t
+
+  val precomputed_values : Precomputed_values.t
+
+  val constraint_constants : Genesis_constants.Constraint_constants.t
+
+  val consensus_constants : Consensus.Constants.t
+end
+
 type Structured_log_events.t += Bootstrap_complete [@@deriving register_event]
 
 val run :
-     logger:Logger.t
+     context:(module CONTEXT)
   -> trust_system:Trust_system.t
```

Comment:
I think even more of this can be moved into context (eg `trust_system`, `verifier`, `network`, probably the persistent frontier handles as well). Doesn't have to happen in this PR, but curious if you agree.

Reply:
I totally agree. I was tempted to do it as I went, but decided not to add more changes in favour of getting something out.